packages/contracts-bedrock/src/L1/OPContractsManager.sol

```diff
@@ -1174,6 +1204,276 @@ contract OPContractsManagerDeployer is OPContractsManagerBase {
+        );
+
+        // Select the correct game type based on the type being used by the existing portals.
+        GameType newGameType;
+        if (_input.usePermissionlessGame) {
+            newGameType = GameTypes.SUPER_CANNON;
+        } else {
+            newGameType = GameTypes.SUPER_PERMISSIONED_CANNON;
+        }
+
+        // Initialize the new AnchorStateRegistry.
+        upgradeToAndCall(
+            _input.opChainConfigs[0].proxyAdmin,
+            address(newAnchorStateRegistry),
+            getImplementations().anchorStateRegistryImpl,
+            abi.encodeCall(
+                IAnchorStateRegistry.initialize,
+                (portals[0].superchainConfig(), newDisputeGameFactory, _input.startingAnchorRoot, newGameType)
+            )
+        );
+
+        // Migrate each portal to the new ETHLockbox and AnchorStateRegistry.
+        for (uint256 i = 0; i < portals.length; i++) {
+            // Authorize the existing ETHLockboxes to use the new ETHLockbox.
+            IETHLockbox existingLockbox = IETHLockbox(payable(address(portals[i].ethLockbox())));
+            newEthLockbox.authorizeLockbox(existingLockbox);
+
+            // Migrate the existing ETHLockbox to the new ETHLockbox.
+            existingLockbox.migrateLiquidity(newEthLockbox);
+
+            // Migrate the portal to the new ETHLockbox and AnchorStateRegistry.
+            portals[i].migrateToSuperRoots(newEthLockbox, newAnchorStateRegistry);
+        }
+
+        // Separate context to avoid stack too deep.
+        {
+            // Deploy a new DelayedWETH proxy for the permissioned game.
+            IDelayedWETH newPermissionedDelayedWETHProxy = IDelayedWETH(
+                payable(
+                    deployProxy({
+                        _l2ChainId: block.timestamp,
+                        _proxyAdmin: _input.opChainConfigs[0].proxyAdmin,
+                        _saltMixer: reusableSaltMixer(_input.opChainConfigs[0]),
+                        _contractName: "DelayedWETH-Interop-Permissioned"
+                    })
+                )
+            );
+
+            // Initialize the new DelayedWETH proxy.
+            upgradeToAndCall(
+                _input.opChainConfigs[0].proxyAdmin,
+                address(newPermissionedDelayedWETHProxy),
+                getImplementations().delayedWETHImpl,
+                abi.encodeCall(IDelayedWETH.initialize, (proxyAdminOwner, portals[0].superchainConfig()))
+            );
+
+            // Deploy the new SuperPermissionedDisputeGame.
+            // NOTE that we use a chain id of 0 here (instead of the block timestamp) because the
+            // use of the chain id is different and actually passed into the constructor of the
+            // dispute game contracts. Since these are Super dispute games and involve multiple
+            // chains, the contracts enforce that the chain id is zero.
+            ISuperPermissionedDisputeGame newSuperPDG = ISuperPermissionedDisputeGame(
```

Comment:
New dispute games cost ~5MM.

The current OP, Soneium and Ink upgrade to v2.0.0 costs 28MM. I'd be quite concerned about the feasibility of running this one more than 3 chains.

I'd suggest adding a gas cost assertion to `test_migrate_withPermissionlessGame_succeeds`.

Reply:
I'll add an assertion. It's only one dispute game total, let's find out what the max is.

Reply:
Yes. Adding an assertion to the tests but the gas usage appears to be ~12m for two chains and it doesn't increase significantly for each additional chain.